src/utils/infer-parser.js

```diff
@@ -93,14 +94,24 @@ function getLanguageByIsSupported(languages, file) {
     return;
   }
 
-  /*
-  We can't use `url.fileURLToPath` here since this module should work in browsers too
-  `URL#pathname` won't work either since `new URL('file:///C:/path/to/file').pathname`
-  equals to `/C:/path/to/file`, try to improve this part in future
-  */
-  file = String(file);
+  if (
+    (typeof URL !== "undefined" && // In `node:vm`, `URL` is not available
```

Comment:
So it would not work when using `import { URL } from 'node:url'`?

Reply:
[`a72201e` (#17490)](https://github.com/prettier/prettier/pull/17490/commits/a72201e4a4e847c8e3ed1bf21f41200c23db2601)